src/Common/tests/System/PlatformDetection.cs

```diff
@@ -34,6 +34,58 @@ public static class PlatformDetection
 
         public static int WindowsVersion { get; } = GetWindowsVersion();
 
+        private static int s_isWinRT = -1;
```

Comment:
@tarekgh, that's not possible nor would it fix the problem.  The issue is that a ```Nullable<T>``` is multiple fields and is not guaranteed to be written/read atomically.